merlin/models/tf/predictions/dot_product.py

```diff
@@ -0,0 +1,371 @@
+import logging
+from typing import List, Optional, Sequence, Union
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+
+from merlin.models.tf.core.prediction import Prediction
+from merlin.models.tf.metrics.topk import AvgPrecisionAt, MRRAt, NDCGAt, PrecisionAt, RecallAt
+from merlin.models.tf.predictions.base import ContrastivePredictionBlock
+from merlin.models.tf.predictions.sampling.base import Items, ItemSampler, ItemSamplersType
+from merlin.models.tf.typing import TabularData
+from merlin.models.tf.utils import tf_utils
+from merlin.models.tf.utils.tf_utils import call_layer, rescore_false_negatives
+from merlin.models.utils import schema_utils
+from merlin.models.utils.constants import MIN_FLOAT
+from merlin.schema import Schema, Tags
+
+LOG = logging.getLogger("merlin_models")
```

Comment:
🗒️  this doesn't need to be addressed in this PR. This is an observation that we should address the inconsistency we have in the name we used inside `getLogger` across the repo:

https://github.com/NVIDIA-Merlin/models/search?p=1&q=getLogger

Reply:
Thanks for the catch! I just opened a new issue for tracking